packages/date-picker/src/vaadin-lit-date-picker-overlay.js

```diff
@@ -0,0 +1,63 @@
+/**
+ * @license
+ * Copyright (c) 2016 - 2023 Vaadin Ltd.
+ * This program is available under Apache License Version 2.0, available at https://vaadin.com/license/
+ */
+import { html, LitElement } from 'lit';
+import { DirMixin } from '@vaadin/component-base/src/dir-mixin.js';
+import { PolylitMixin } from '@vaadin/component-base/src/polylit-mixin.js';
+import { OverlayMixin } from '@vaadin/overlay/src/vaadin-overlay-mixin.js';
+import { PositionMixin } from '@vaadin/overlay/src/vaadin-overlay-position-mixin.js';
+import { overlayStyles } from '@vaadin/overlay/src/vaadin-overlay-styles.js';
+import { ThemableMixin } from '@vaadin/vaadin-themable-mixin/vaadin-themable-mixin.js';
+import { datePickerOverlayStyles } from './vaadin-date-picker-overlay-styles.js';
+
+/**
+ * An element used internally by `<vaadin-date-picker>`. Not intended to be used separately.
+ *
+ * @extends HTMLElement
+ * @mixes PositionMixin
+ * @mixes OverlayMixin
+ * @mixes DirMixin
+ * @mixes ThemableMixin
+ * @private
+ */
+class DatePickerOverlay extends PositionMixin(OverlayMixin(DirMixin(ThemableMixin(PolylitMixin(LitElement))))) {
+  static get is() {
+    return 'vaadin-date-picker-overlay';
+  }
+
+  static get styles() {
+    return [overlayStyles, datePickerOverlayStyles];
+  }
+
+  static get properties() {
+    return {
+      /**
+       * When true, the overlay is visible and attached to body.
+       * This property config is overridden to set `sync: true`.
+       */
+      opened: {
+        type: Boolean,
+        notify: true,
+        observer: '_openedChanged',
+        reflectToAttribute: true,
+        sync: true,
+      },
```

Comment:
The whole thing with overriding the property is needed because of this method:

https://github.com/vaadin/web-components/blob/55364e0d8f6629199886d87075b797586d82c98c/packages/date-picker/src/vaadin-date-picker-mixin.js#L505-L507

When converting to `LitElement`, this property would become async and as a result, timings would be off e.g. when confirming date on <kbd>Enter</kbd> (which caused validation to be called twice and fail one of the tests).

Adding `sync: true` here ensures the current synchronous opening / closing behavior is preserved.